osquery/sql/sqlite_util.cpp

```diff
@@ -195,6 +200,20 @@ void SQLiteDBInstance::addAffectedTable(VirtualTableContent* table) {
   affected_tables_.insert(std::make_pair(table->name, table));
 }
 
+TableAttributes SQLiteDBInstance::getAttributes() const {
+  const SQLiteDBInstance* rdbc = this;
+  if (isPrimary() && !managed_) {
```

Comment:
This is weird. What happens is everytime a DB instance is requested, a new object is provided. That object internally knows if it is "backed" by "primary" virtual table instances or "ephemeral". If there's ever contention a new DB and set of virtual table instances are returned.

This forwarding is required because internally to the virtual table implementations, there are "callbacks" to a specific instance here. We can access that with a `::getConnection(true)`.